ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -65,10 +65,13 @@ openMempool registry chainDB cfg tracer = do
 -- | Internal state in the mempool
 data InternalState blk = IS {
       -- | Transactions currently in the mempool
-      isTxs :: TxSeq (GenTx blk)
+      isTxs          :: TxSeq (GenTx blk)
 
       -- | The tip of the chain that 'isTxs' was validated against
-    , isTip :: ChainHash blk
+    , isTip          :: ChainHash blk
+
+      -- | The mempool 'TicketNo' counter
+    , isLastTicketNo :: TicketNo
```

Comment:
Maybe refer to (the docstring of) `vrLastTicketNo` here, so that the reader knows that the 0th ticket is never used for a transaction (important for `getTxsAfter`) and that this number is incremented before using it for a transaction.